### PR TITLE
feat(ProgressStepper): apply tokens

### LINKED ISSUE #6235: Consume tokens in: Progress Stepper
Add tokens to progress stepper. Figma design at https://www.figma.com/file/VMEX8Xg2nzhBX8rfBx53jp/Penta-Component-Library?type=design&node-id=7644%3A40986&mode=dev

### PR DESCRIPTION
**Pull request by srambach**
Fixes #6235 

Notes:

- Removed `.pf-m-info` from making the current step have the icon be blue. Added `.pf-m-in-progress-alt` _not_ to turn the current icon blue.
- Added `.pf-m-custom` to complete the other states that are provided.

### CODE REVIEW

File: src/patternfly/components/ProgressStepper/progress-stepper.scss [hunk 1]
```diff
@@ -138,49 +139,51 @@ $pf-v5-c-progress-stepper--breakpoint-map: build-breakpoint-map();
...
-  --#{$progress-stepper}__step-title--m-help-text--TextDecorationThickness: var(--#{$pf-global}--BorderWidth--sm);
+  --#{$progress-stepper}__step-title--m-help-text--TextDecorationColor: currentcolor;
+  --#{$progress-stepper}__step-title--m-help-text--TextDecorationThickness: var(--pf-t--global--border--width--box--default);
   --#{$progress-stepper}__step-title--m-help-text--TextUnderlineOffset: #{pf-size-prem(4px)};
-  --#{$progress-stepper}__step-title--m-help-text--hover--TextDecorationColor: var(--#{$pf-global}--Color--100);
-  --#{$progress-stepper}__step-title--m-help-text--focus--TextDecorationColor: var(--#{$pf-global}--Color--100);
-  --#{$progress-stepper}__step-title--m-help-text--hover--Color: var(--#{$pf-global}--Color--100);
-  --#{$progress-stepper}__step-title--m-help-text--focus--Color: var(--#{$pf-global}--Color--100);
+  --#{$progress-stepper}__step-title--m-help-text--hover--Color: var(--pf-t--global--text--color--brand--hover);
+  --#{$progress-stepper}__step-title--m-help-text--focus--Color: var(--pf-t--global--text--color--brand--hover);
```
Comment (thatblindgeye) [inline src/patternfly/components/ProgressStepper/progress-stepper.scss:177]: Nit: could remove the focus var and use the hover var for hover and focus

File: src/patternfly/components/ProgressStepper/progress-stepper.scss [hunk 2]
```diff
@@ -138,49 +139,51 @@ $pf-v5-c-progress-stepper--breakpoint-map: build-breakpoint-map();
...
 
   // Help text variables the step title for failure state
-  --#{$progress-stepper}__step--m-danger__step-title--m-help-text--hover--Color: var(--#{$pf-global}--danger-color--200);
-  --#{$progress-stepper}__step--m-danger__step-title--m-help-text--focus--Color: var(--#{$pf-global}--danger-color--200);
-  --#{$progress-stepper}__step--m-danger__step-title--m-help-text--TextDecorationColor: var(--#{$pf-global}--danger-color--100);
-  --#{$progress-stepper}__step--m-danger__step-title--m-help-text--hover--TextDecorationColor: var(--#{$pf-global}--danger-color--200);
-  --#{$progress-stepper}__step--m-danger__step-title--m-help-text--focus--TextDecorationColor: var(--#{$pf-global}--danger-color--200);
+  --#{$progress-stepper}__step--m-danger__step-title--m-help-text--hover--Color: var(--pf-t--global--text--color--status--danger--hover);
+  --#{$progress-stepper}__step--m-danger__step-title--m-help-text--focus--Color: var(--pf-t--global--text--color--status--danger--hover);
```
Comment (thatblindgeye) [inline src/patternfly/components/ProgressStepper/progress-stepper.scss:181]: Same as above

File: src/patternfly/components/ProgressStepper/progress-stepper.scss [hunk 3]
```diff
@@ -286,34 +289,40 @@ $pf-v5-c-progress-stepper--breakpoint-map: build-breakpoint-map();
...
     --#{$progress-stepper}__step-title--FontWeight: var(--#{$progress-stepper}__step--m-current__step-title--FontWeight);
     --#{$progress-stepper}__step-title--Color: var(--#{$progress-stepper}__step--m-current__step-title--Color);
+    --#{$progress-stepper}__step-icon--Color: var(--#{$progress-stepper}__step--m-current__step-icon--Color);
   }
 
   &.pf-m-pending {
     --#{$progress-stepper}__step-title--Color: var(--#{$progress-stepper}__step--m-pending__step-title--Color);
   }
 
+  &.pf-m-in-progress-alt {
```
Comment (thatblindgeye) [inline src/patternfly/components/ProgressStepper/progress-stepper.scss:299]: Do we need the `alt` suffix for this modifier?

**Review by mcoker — COMMENTED** [review]
Just a couple of things, otherwise lgtm!

**Comment by srambach** [comment]
OK, on the name of the in progress alt variation... here's the dilemma. 
- The actual icon used as an _alternative_ in this variation is the pf-in-progress icon.
- The modifier `.pf-m-in-progress` elsewhere means we are going to put the spinner in the thing.
- The text on that step says it's "in process."

So using `.pf-m-in-progress` doesn't seem right because it has nothing to do with the spinner. `.pf-m-in-process` seems too close so it seems like a mistake and doesn't match the icon. 

I like the idea of using that variant to turn any icon back to not-a-color IF we really need this. And then we need a name, because `.pf-m-plain` usually relates to outlines and backgrounds... @lboehling @andrew-ronaldson is this really a thing that needs to be? Is there a reasonable use case where we just can't have the current step have the blue color?

@mcoker @thatblindgeye

**Review by mcoker — APPROVED** [review]

**Review by thatblindgeye — APPROVED** [review]
Naming of the progress alt class isn't a blocker for this PR, so otherwise lgtm

**Comment by andrew-ronaldson** [comment]
Screenshot 2024-01-29 at 10 53 37 AM
Can you please remove the 4px spacer from the top. It looks like it is aligned bottom with the step indicator

**Review by mcoker — COMMENTED** [review]
Also noticed there is some extra inline padding being added to the help text `<button>`

Screenshot 2024-01-29 at 9 59 32 AM

**Review by thatblindgeye — APPROVED** [review]

**Review by mcoker — APPROVED** [review]
🍰

**Review by andrew-ronaldson — APPROVED** [review]
Sweeeeeeeeet stuff

**Pull request merged**